File: 20-valid-parentheses/20-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */

const isValid = (s) => {
  const stack = [];
  const map = {
    '(':')',
    '{':'}',
    '[':']'
  };
  for (let char of s) {
    if (char === '(' || char === '{' || char === '[') {
      stack.push(char);
    } else if (map[stack.pop()] !== char) {
      return false;
    }
  }
  return !stack.length;
}

File: 20-valid-parentheses/20-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        parens = {'(':')', '[':']', '{':'}'}
        stack = []
        for char in s:
            if char in parens:
                stack.append(char)
            elif len(stack) is 0 or parens[stack.pop()] != char:
                return False
        return len(stack) == 0
            
                
                
    